dotcom-rendering/src/web/components/Island.tsx

```diff
@@ -7,7 +7,7 @@ type When = 'idle' | 'visible';
 interface HydrateProps {
 	deferUntil?: When;
 	clientOnly?: false;
-	placeholderHeight?: undefined;
+	placeholderHeight?: never;
```

Comment:
Would removing unusable props make more sense?

```ts
interface ServerSideRenderedProps {
	children: JSX.Element;
}

interface ClientOnlyProps {
	clientOnly: true;
	deferUntil?: When;
	placeholderHeight?: number;
	children: JSX.Element;
}

type Props = ServerSideRenderedProps | ClientOnlyProps
```

Reply:
This wouldn’t work with Typescript, as we need overlap between the two types to set it as the props of Island’s first de-structured argument.

Reply:
Yeah. I actually think `never` is kind of lovely. I mean, it would be _better_ to just not have the prop but TS doesn't like that so in this case Max's suggestion makes so much more sense than what was there before